src/mame/konami/konamigv.cpp

```diff
@@ -834,13 +1059,28 @@ void tokimeki_state::tokimeki_serial_w(offs_t offset, uint16_t data, uint16_t me
 		m_serial_len++;
 	}
 
-	m_serial_clk = clk;
+	m_serial_clk = serial_clk;
 }
 
 void tokimeki_state::tmosh(machine_config &config)
 {
 	konamigv(config);
 	m_maincpu->set_addrmap(AS_PROGRAM, &tokimeki_state::tmosh_map);
+
+	auto &screen(SCREEN(config, "printer", SCREEN_TYPE_RASTER));
+	screen.set_size(PRINTER_PAGE_WIDTH, PRINTER_PAGE_HEIGHT);
+	screen.set_visarea_full();
+	screen.set_refresh_hz(10); // infrequently updated, only displays printed page
+	screen.set_screen_update(FUNC(tokimeki_state::printer_screen_update));
+}
+
+uint32_t tokimeki_state::printer_screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
+{
+	if (!m_printer_is_printing)
+		return 0;
+
+	copybitmap(bitmap, m_page_bitmap, 0, 0, 0, 0, cliprect);
+	return 0;
 }
```

Comment:
You may be able to get a slight performance win by:
* Set a flag on reset and when you actually update the printer bitmap.
* If the flag is not set, just `return UPDATE_HAS_NOT_CHANGED;` here.
* If the flag is set, update the bitmap, clear the flag, and return zero here.

Reply:
Done. https://github.com/mamedev/mame/pull/12233/commits/0614bd38a9c0fe08aa02877324319174a37d434c